jinja2/environment.py

```diff
@@ -335,6 +336,11 @@ def __init__(self,
         self.enable_async = enable_async
         self.is_async = self.enable_async and have_async_gen
 
+        self.native = native
+        if self.native:
+            self.code_generator_class = NativeCodeGenerator
```

Comment:
I don't like this. I'd rather add a `native_code_generator_class` class attribute and pick this one here. Otherwise there's no way of using a custom native code generator class without setting it after instantiating the environment (instead of setting it on the class level on a subclass)